### ISSUE TITLE
Flag unnecessary type conversions

### ISSUE THREAD

**Issue by Jacalz** [dominikh/go-tools#936]
At the moment, staticcheck does not seem to be flagging any unnecessary type conversions. I have seen one or two lately. I don't know if it is a bug or that there simply isn't a check for it, but I assume the latter.

Incorrect code:
```go
func print(a int64) {
    fmt.Println(a)
}

func main() {
   a := int64(2)
   print(int64(a))
}
``` 

Correct code:
```go
func print(a int64) {
    fmt.Println(a)
}

func main() {
   a := int64(2)
   print(a)
}
``` 

Probably related to https://github.com/dominikh/go-tools/issues/128

**Comment by dominikh** [comment]
Non-trivial because of build tags, see https://github.com/mdempsky/unconvert

**Comment by Jacalz** [comment]
Ah, I see. This was just something that I had expected staticcheck to do.

**Comment by dominikh** [comment]
I'm currently working on better support for build tags, after which I may look at this again.
